src/common/k8s.ts

```diff
@@ -0,0 +1,11 @@
+import { CoreV1Api, KubeConfig } from '@kubernetes/client-node'
+
+let apiClient: CoreV1Api
+
+export function getApiClient(): CoreV1Api {
+  if (apiClient) return apiClient
```

Comment:
I am a bit confused by this first line. There is a global variable in this file that only declares the variable. I don't see where it gets set, but then this first line checks if it is set and if it is, then return it?

Reply:
It a way to implement Singleton pattern